include/http_core.h

```diff
@@ -1109,6 +1109,28 @@ AP_DECLARE(int) ap_state_query(int query_code);
  */
 AP_CORE_DECLARE(conn_rec *) ap_create_slave_connection(conn_rec *c);
 
+struct apr_pollfd_t;
+
+AP_DECLARE_HOOK(apr_status_t, get_conn_in_pollfd,
+                (conn_rec *c, struct apr_pollfd_t *pfd,
+                 apr_interval_time_t *ptimeout))
+
+/**
+ * Pass in a `struct apr_pollfd_t*` and get `desc_type` and `desc`
+ * populated with a suitable value for polling connection input.
+ * For primary connection (c->master == NULL), this will be the connection
+ * socket. For secondary connections this may differ or not be available
+ * at all.
+ * Note that APR_NO_DESC may be set to indicate that the connection
+ * input is already closed.
+ *
+ * @param pfd  the pollfd to set the descriptor in
+ * @param ptimeout  != NULL to retrieve the timeout in effect
+ * @return ARP_SUCCESS when the information was assigned.
+ */
+AP_CORE_DECLARE(apr_status_t) ap_get_conn_in_pollfd(conn_rec *c,
+                                      struct apr_pollfd_t *pfd,
+                                      apr_interval_time_t *ptimeout);
```

Comment:
Sorry, name bikeshedding :/
ap_get_conn_in_pollfd() sounds like it's filling a conn_rec from an apr_pollfd_t but it's the other way around IIUC.
Maybe ap_get_pollfd_from_conn() or ap_set_conn_in_pollfd()?

Reply:
I was mentally prepared for that. And I am not emotionally involved in the proposed name. Your propposal works for me.

Reply:
Changed to ap_get_pollfd_from_conn()